src/umpire/ResourceManager.cpp

```diff
@@ -225,6 +272,20 @@ ResourceManager::initialize()
   }
 #endif
 
+  {
+    std::unique_ptr<strategy::AllocationStrategy> allocator{
+      new strategy::FixedPool{"__umpire_0_byte_pool",
+        getNextId(),
+        Allocator{m_allocators_by_name["NULL"]},
+        1}
+    };
+
+    int id{allocator->getId()};
+    m_allocators_by_name["__umpire_0_byte_pool"] = allocator.get();
```

Comment:
Could rename this to include `internal`, and again, figure out how to hide it from users.

Reply:
I think we should include `internal` at the least.

Reply:
A private variable in the class could work, that is returned from `getZeroByteAllocator()`

Reply:
I think registering every allocator is good practice, for now.